x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalVerifier.java

```diff
@@ -38,10 +39,14 @@ public Failures verify(LogicalPlan plan, boolean skipRemoteEnrichVerification) {
             if (failures.hasFailures() == false) {
                 if (p instanceof PostOptimizationVerificationAware pova) {
                     pova.postOptimizationVerification(failures);
+                } else if (p instanceof PostOptimizationPlanVerificationAware popva) {
```

Comment:
Added `PostOptimizationPlanVerificationAware` to have the plan available for validations. This is similar to `PostAnalysisPlanVerificationAware`

Reply:
I think it makes sense. 👍